.github/workflows/pypi-mac.yml

```diff
@@ -0,0 +1,41 @@
+name: Build Mac OS X package and upload it to PyPI
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [macos-latest]
+        architecture: [x64]
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        architecture: ${{ matrix.architecture }}
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools wheel twine
+    - name: Build Python wheel
+      run: |
+        python setup.py bdist_wheel
+    - name: Upload artifacts for inspection
+      uses: actions/upload-artifact@v2
+      with:
+        name: wheel_${{ matrix.os }}_${{ matrix.architecture }}_${{ matrix.python-version }}
+        path: dist/*.whl
+    - name: Publish sdists and wheels to PyPI
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+      run: |
+        twine upload dist/* || :
```

Comment:
I have no idea how to run this step on release only (without copy pasting the entire file with different `on:`. Technically, until the version si bumped, this will simply fail once the existing version is already there, but I am not very satisfied with that.